support-frontend/assets/pages/supporter-plus-landing/supporterPlusLanding.tsx

```diff
@@ -156,147 +208,234 @@ export function SupporterPlusLandingPage({
 		}
 	}, [paymentComplete]);
 
-	return (
-		<PageScaffold
-			id="supporter-plus-landing"
-			header={
-				<>
-					<Header>
-						<Hide from="desktop">
-							<CountrySwitcherContainer>
-								<CountryGroupSwitcher {...countrySwitcherProps} />
-							</CountrySwitcherContainer>
-						</Hide>
-					</Header>
-					<Nav {...countrySwitcherProps} />
-				</>
-			}
-			footer={
-				<FooterWithContents>
-					<FooterLinks></FooterLinks>
-				</FooterWithContents>
+	const buttonContainerRef = useRef(null);
+	const [stripeDisplayed, setStripeDisplayed] = useState(
+		!optimisedMobileLayout2,
+	);
+	const [buttonIsStuck, setButtonIsStuck] = useState(false);
+
+	// Use IntersectionObserver to detect when button is 'stuck' at the bottom
+	// of the viewport. The bottom of the observable area is set to -1px so that
+	// when the button is stuck it is not considered to be fully visible. The
+	// top edge is similarly extended upwards so the button is considered fully
+	// visible when scrolling off the top of the screen.
+
```

Comment:
Is there a reason you opted for this kind of approach over using `position: sticky`? I feel like using primarily CSS for this would make the code a lot simpler to follow and to either clean up or ship following the test, but I don't know if there are some technical restrictions in play here.

Reply:
I took this from manage-frontend (SwitchOptions.tsx) which had a corresponding storybook with an example here -> https://61eee3dabb99df003a4090fd-qlolqzjxkh.chromatic.com/?path=/story/pages-switchoptions--below-threshold

Reply:
Will see if I can research, do a simpler way now. If you know of an article that might help perhaps drop it in chat?

Reply:
Yes, all related isStuck useState/useEffects were redundant in this case. Could of checked this more closely. Have removed it all.